### PR TITLE
Lazy SmartReferenceManager thread creation

### PR DESCRIPTION
**Pull request by alxtt**
Поток, создаваемый в SmartReferenceManager не нужен вообще.
Для обратной совместимости есть возможность включить его (см. конструктор ObjectObjectTroveStorage).

### CODE REVIEW

File: mxcache-runtime/src/main/java/com/maxifier/mxcache/impl/caches/def/ObjectObjectTroveStorage.java [hunk 1]
```diff
@@ -14,6 +16,11 @@
  */
 public class ObjectObjectTroveStorage<K> extends THashMap<K, Object> implements ObjectObjectStorage<K> {
 
+    public ObjectObjectTroveStorage() {
+        super();
```
Comment (akochurov) [inline mxcache-runtime/src/main/java/com/maxifier/mxcache/impl/caches/def/ObjectObjectTroveStorage.java]: What is this super() call for? It's redundant...

**Comment by akochurov** [comment]
LGTM

File: mxcache-runtime/src/main/java/com/maxifier/mxcache/impl/caches/def/ObjectObjectTroveStorage.java (hunk 1 above)
Comment (alxtt) [inline mxcache-runtime/src/main/java/com/maxifier/mxcache/impl/caches/def/ObjectObjectTroveStorage.java]: implicit code- bad code...

Comment (klya) [inline mxcache-runtime/src/main/java/com/maxifier/mxcache/impl/caches/def/ObjectObjectTroveStorage.java]: Better developer writes less.

**Comment by klya** [comment]
LGTM. But it's better for me to start thread in SmartReferenceManager#getReferenceQueue.

**Comment by alxtt** [comment]
> But it's better for me to start thread in SmartReferenceManager#getReferenceQueue

This will destroy the optimization - getReferenceQueue called ALWAYS.

**Comment by klya** [comment]
If it is called always than it is used always.

**Comment by klya** [comment]
Stop! Cleaner thread required by WeakList.InstanceReference and CleanableRegister#registerInstance.

**Comment by klya** [comment]
:-1:

**Pull request closed without merging**